snuba/cli/dlq_manager.py

```diff
@@ -0,0 +1,151 @@
+    "limit",
+    type=int,
+    help="Max number of messages to display, 10 by default",
+)
+def list(storage_set: str, offset: Optional[int], limit: Optional[int]) -> None:
+    """
+    List all messages found in a dead-letter topic
+    """
+    limit = limit or 10
+    offset = offset or 0
+    messages = _consume_dead_letters(storage_set, offset, limit)
+    line_break = "-" * 50
+    if messages:
+        click.echo(
+            f"\nListing the first {limit} messages from "
+            f"offset {offset} in {storage_set} dead-letter topic:"
+            f"\n(see --help for pagination)\n"
+        )
+        for message in messages:
+            click.echo(line_break)
+            click.echo(
+                {
+                    "offset": message.offset,
+                    "invalid_message": message.payload.value,
+                }
+            )
+    else:
+        click.echo(f"\nNo messages found in {storage_set} dead-letter topic!")
+    click.echo(line_break)
+
+
+def _consume_dead_letters(
+    storage_set: str, offset: int, limit: int
+) -> Sequence[Message[KafkaPayload]]:
+    dead_letter_topic_arroyo = ArroyoTopic(
+        STORAGES_WITH_DLQ[StorageSetKey(storage_set)].value
+    )
+    consumer = _build_consumer(storage_set)
+    messages: MutableSequence[Message[KafkaPayload]] = []
+
+    if offset != 0:
+        try:
+            message = consumer.poll(10)
+            consumer.tell()
+            consumer.seek({Partition(dead_letter_topic_arroyo, 0): offset})
+        except Exception as e:
+            click.echo(f"An error occured: {e}")
+            consumer.close()
+            return messages
+
+    for _ in range(limit):
+        try:
+            message = consumer.poll(10)
+            if message is None:
+                consumer.close()
+                return messages
+            messages.append(message)
+        except Exception as e:
+            click.echo(f"An error occured: {e}")
+            consumer.close()
+            return messages
+    return messages
+
+
+@dlq_manager.command()
+@click.option(
+    "--storage",
+    "storage_set",
+    type=click.Choice(
+        [storage_set_key.value for storage_set_key in STORAGES_WITH_DLQ.keys()]
+    ),
+    help="The storage set to see info about dead-letter topic for",
+    required=True,
+)
+def info(storage_set: str) -> None:
+    """
+    Display useful info for a dead-letter topic for a storage set
+    """
+    earliest_offset, latest_offset = _get_offsets_info(storage_set)
+    dead_letter_topic_snuba = STORAGES_WITH_DLQ[StorageSetKey(storage_set)]
+    click.echo(f"\nDisplaying info for {storage_set}:\n")
+    click.echo(f"Dead letter topic name: {dead_letter_topic_snuba.value}")
+    click.echo(f"Earliest offset: {earliest_offset}")
+    click.echo(f"Latest offset: {latest_offset}")
+
+
+def _get_offsets_info(storage_set: str) -> Tuple[Optional[int], Optional[int]]:
+    consumer = _build_consumer(storage_set)
+    message = consumer.poll(10)
```

Comment:
Don't these functions take too long to run if the DLQ is empty?

Reply:
Yeah it takes too long to run, but when this timeout is less than 10 it's really flakey. I'm not sure why, it sometimes just doesn't get a message even though the topic has messages.

Reply:
I see, I'd probably print a message for the user if it's waiting up to 10 seconds